assets/src/blocks/Timeline/TimelineNode.js

```diff
@@ -0,0 +1,52 @@
+import { useScript } from './useScript';
+import { useStyleSheet } from './useStyleSheet';
+import { useRef, useEffect } from 'react';
+import { uniqueId } from 'lodash';
+
+const TIMELINE_JS_VERSION = '3.6.3';
+
+export const TimelineNode = (props) => {
+	const {
+		google_sheets_url,
+		timenav_position,
+		start_at_end,
+		language
+	} = props;
+
+	const timelineNode = useRef(null);
+
+	useStyleSheet(
+    `https://cdn.knightlab.com/libs/timeline3/${TIMELINE_JS_VERSION}/css/timeline.css`
```

Comment:
I assume that in a lot of cases the timeline will be above the fold. Only starting the requests for the stylesheet and js files at this point will definitely cause the block to start displaying too late and couldcause a layout jump. Compared to including the sheet in the html doc, it should definitely be noticeable.

We could still include this stylesheet and the js with the backend if a post has the block in it.

Reply:
After checking around looks like mostly the block is used well below the fold. So maybe loading it in here only is preferable after all. It would also delay the rendering of the timeline, which can be pretty intensive.

Reply:
Yes, so I did ran some tests, which were not super conclusive as each time you ran them you get different scores for performance..., but what is consistent is that if we add the files as we do now (enqueue from PHP), Lighthouse complains about them blocking the main thread:

![image](https://user-images.githubusercontent.com/340766/95581023-a6d92280-0a0e-11eb-91d6-bd0c5faf4e30.png)

And the timeline gets listed under "blocking" resources here:
![image](https://user-images.githubusercontent.com/340766/95581124-d425d080-0a0e-11eb-9680-2cc040dbfdac.png)

In this version (load on render using hooks), those issues disappear and you can see the timeline being loaded after the "first contentful paint", "largest contentful pain", and "DOM Content loaded" events:

![image](https://user-images.githubusercontent.com/340766/95581248-10f1c780-0a0f-11eb-81f9-19ee629929e4.png)

In the process I found out the author's (KnightLab) CDN is not gzipping the files, so I'll see if these exists in another CDN with compression:

![image](https://user-images.githubusercontent.com/340766/95581328-2f57c300-0a0f-11eb-894c-b4f1795eb48d.png)
 
As for the jumpyness, this component has a fixed height (luckily), so we can make it render something to fill up the space and look better.

Reply:
Good catch about the compression :100: Probably makes most sense if we put it on our cloudflare, then that's one less domain to connect to.

Reply:
cdnjs (that we already use for other libs), seems to have it

https://cdnjs.com/libraries/timelinejs